BACKLOG-17979 - fixed memory leak bug for NamedClusterManager by removing cache for non-Embedded MetaStores

pentaho-jrice opened the pull request.

pentaho-jrice commented:
@rmansoor, @tkafalas 

Please take a look and review

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterManager.java
  impl/cluster/src/test/java/org/pentaho/big/data/impl/cluster/NamedClusterManagerTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.big.data.impl.cluster.NamedClusterManager
 * Branch Change: **-11.11%**:small_red_triangle_down:
 * Complexity Change: **-3.70%**:small_red_triangle_down:
 * Instruction Change: **-5.15%**:small_red_triangle_down:
 * Line Change: **-4.05%**:small_red_triangle_down:

#### Checkstyle violations

The code submitted is not complying with one or more of our guidelines. To perform these checks locally, [please use this tool](https://github.com/pentaho/pentaho-coding-standards).

###### /home/buildguy/project/impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterManager.java

> :heavy_exclamation_mark: '(' is not followed by whitespace. at Line: 101 Column: 40

> :heavy_exclamation_mark: ')' is not preceded with whitespace. at Line: 101 Column: 109

wingman-pentaho commented:
### Build Completed

:x: This pull request has errors. They will need to be addressed before it can be accepted. See below for more details. Some links are also available below for further assistance in addressing those issues.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterManager.java
  impl/cluster/src/test/java/org/pentaho/big/data/impl/cluster/NamedClusterManagerTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.big.data.impl.cluster.NamedClusterManager
 * Branch Change: **-11.11%**:small_red_triangle_down:
 * Complexity Change: **-3.70%**:small_red_triangle_down:
 * Instruction Change: **-5.15%**:small_red_triangle_down:
 * Line Change: **-4.05%**:small_red_triangle_down:

#### Checkstyle violations

The code submitted is not complying with one or more of our guidelines. To perform these checks locally, [please use this tool](https://github.com/pentaho/pentaho-coding-standards).

###### /home/buildguy/project/impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterManager.java

> :heavy_exclamation_mark: '(' is not followed by whitespace. at Line: 101 Column: 40

> :heavy_exclamation_mark: ')' is not preceded with whitespace. at Line: 101 Column: 109

pentaho-jrice commented:
@rmansoor , @tkafalas 

Updated PR with fixes for Wingman.  Fixed checkstyle issues.  Expanded coverage.  Simplified logic in getMetaStoreFactory() method to decrease complexity.

wingman-pentaho commented:
### Build Completed

:white_check_mark: This pull request has passed all validations.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterManager.java
  impl/cluster/src/test/java/org/pentaho/big/data/impl/cluster/NamedClusterManagerTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.big.data.impl.cluster.NamedClusterManager
 * Branch Change:  + 5.56%
 * Complexity Change:  + 3.70%
 * Instruction Change:  + 2.99%
 * Line Change:  + 1.43%

wingman-pentaho commented:
### Build Completed

:white_check_mark: This pull request has passed all validations.

#### Build Commands

```sh
mvn -B -fn -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd clean install && mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd site
```

#### Cleanup Commands
```sh
mvn -B -f 'pom.xml' -pl 'impl/cluster' -P '!assemblies' -amd build-helper:remove-project-artifact
```

#### Changed files

```console
  impl/cluster/src/main/java/org/pentaho/big/data/impl/cluster/NamedClusterManager.java
  impl/cluster/src/test/java/org/pentaho/big/data/impl/cluster/NamedClusterManagerTest.java
```

#### Unit test coverage change

These statistics help you identify how your changes have affected the coverage of the following files. If a file is not in this list, then its coverage was not affected by your changes. To get some help interpreting these metrics, please refer to [Jacoco's documentation](http://eclemma.org/jacoco/trunk/doc/counters.html).

###### org.pentaho.big.data.impl.cluster.NamedClusterManager
 * Branch Change:  + 5.56%
 * Complexity Change:  + 3.70%
 * Instruction Change:  + 2.99%
 * Line Change:  + 1.43%
